sentinel-extension/sentinel-datasource-redis/src/main/java/com/alibaba/csp/sentinel/datasource/redis/RedisDataSource.java

```diff
@@ -0,0 +1,82 @@
+package com.alibaba.csp.sentinel.datasource.redis;
+
+import com.alibaba.csp.sentinel.datasource.AbstractDataSource;
+import com.alibaba.csp.sentinel.datasource.ConfigParser;
+import com.alibaba.csp.sentinel.log.RecordLog;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.pubsub.RedisPubSubAdapter;
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection;
+import io.lettuce.core.pubsub.api.sync.RedisPubSubCommands;
+
+
+/**
+ * Redis data resource. when data publish in redis,this will update rule config in time
+ *
+ * @author moon tiger
+ */
+
+public class RedisDataSource<T> extends AbstractDataSource<String, T> {
+
+    private RedisClient redisClient = null;
+
+    private String ruleKey;
+
+    public RedisDataSource(RedisClient client, String ruleKey, String channel, ConfigParser<String, T> parser) {
+        super(parser);
+        this.redisClient = client;
+        this.ruleKey = ruleKey;
+        initConfig();
+        subscribeFromChannel(channel);
+    }
+
+    private void subscribeFromChannel(String channel) {
+        StatefulRedisPubSubConnection<String, String> pubSubConnection = redisClient.connectPubSub();
+        RedisPubSubAdapter<String, String> adapterListener = new DelegatingRedisPubSubListener();
+        pubSubConnection.addListener(adapterListener);
+        RedisPubSubCommands<String, String> sync = pubSubConnection.sync();
+        sync.subscribe(channel);
+    }
+
+
+    private void initConfig() {
+        try {
+            loadConfig();
```

Comment:
Here you need to load the initial config and update the corresponding property like this (referenced from other datasource implementation):

```java
private void loadInitialConfig() {
    try {
        T newValue = loadConfig();
        if (newValue == null) {
            RecordLog.warn("[RedisDataSource] WARN: initial config is null, you may have to check your data source");
        }
        getProperty().updateValue(newValue);
    } catch (Exception ex) {
        RecordLog.warn("[RedisDataSource] Error when loading initial config", ex);
    }
}
```